overlord/snapstate/agentnotify/agentnotify.go

```diff
@@ -77,3 +76,30 @@ var sendClientFinishRefreshNotification = func(snapsup *snapstate.SnapSetup) {
 		}
 	}()
 }
+
+var sendClientFinishRefreshNotification = func(st *state.State, snapsup *snapstate.SnapSetup) {
```

Comment:
This helper is not covered, `TestMaybeAsyncPendingRefreshNotification` and `TestMaybeAsyncPendingRefreshNotificationSkips` are good examples.

Reply:
Also, I think we could also have a common helper for finished/pending refreshes because most code is shared , but maybe in a follow up PR. I believe it is good in its current form.

Reply:
Done.